### PR TITLE
Guarantee `char` layout

### PR DESCRIPTION
**Pull request by joshlf**

### CODE REVIEW

**Comment by joshlf** [comment]
Ping 🙂

**Comment by ehuss** [comment]
Nominated to get lang's attention.

File: src/types/textual.md [hunk 1]
```diff
@@ -17,7 +17,9 @@ is valid UTF-8. Calling a `str` method with a non-UTF-8 buffer can cause
 Since `str` is a [dynamically sized type], it can only be instantiated through a
 pointer type, such as `&str`.
 
-## Bit validity
+## Layout and bit validity
+
+`char` is guaranteed to have the same size and alignment as `u32` on all platforms.
```
Comment (joshtriplett) [inline src/types/textual.md:22]:
```suggestion
`char` is guaranteed to have the same size and alignment as `u32` on all platforms.

`Option<char>` is guaranteed to have the same size and alignment as `char` on all platforms.
```

(Unless we want to put this somewhere else.)

Comment (joshtriplett) [reply src/types/textual.md:22]: This suggestion should probably be handled separately, not in this PR.

**Comment by joshtriplett** [comment]
Consensus from lang is to merge.

std docs should also be updated.

File: src/types/textual.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (joshlf) [reply src/types/textual.md:22]: No opposition to this going in `core` as well, but we already have text in the Reference about the sizes of primitive types: https://doc.rust-lang.org/reference/type-layout.html#primitive-data-layout

[the remark this answers is not in the document]
Comment (joshlf) [reply src/types/textual.md:22]: Put up a PR to add the same text to `core`: https://github.com/rust-lang/rust/pull/116894

**Comment by WaffleLapkin** [comment]
What's the status here?

File: src/types/textual.md (hunk 1 above)
Comment (joshlf) [reply src/types/textual.md:22]: Filed https://github.com/rust-lang/reference/issues/1426 to track this.

**Pull request merged**